### PR TITLE
daily indices

### PR DESCRIPTION
**Pull request by pawelszymczyk**

### CODE REVIEW

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java [hunk 1]
```diff
@@ -77,4 +80,63 @@ protected XContentBuilder notEndedDocument(MessageMetadata message, long timesta
                 .field(CLUSTER, clusterName);
     }
 
+    public static class Builder {
```
Comment (awislowski) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java]: I wonder if not give more descriptive name for builder 
and extract it for using it also in  ...ava/pl/allegro/tech/hermes/tracker/elasticsearch/frontend/ElasticsearchLogRepository.java with just few more params

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java [hunk 2]
```diff
@@ -9,28 +9,31 @@
...
 import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
-import static pl.allegro.tech.hermes.api.SentMessageTraceStatus.*;
+import static pl.allegro.tech.hermes.api.SentMessageTraceStatus.DISCARDED;
+import static pl.allegro.tech.hermes.api.SentMessageTraceStatus.FAILED;
+import static pl.allegro.tech.hermes.api.SentMessageTraceStatus.INFLIGHT;
+import static pl.allegro.tech.hermes.api.SentMessageTraceStatus.SUCCESS;
 import static pl.allegro.tech.hermes.tracker.elasticsearch.DocumentBuilder.build;
-import static pl.allegro.tech.hermes.tracker.elasticsearch.LogSchemaAware.TypedIndex.SENT_MESSAGES;
 
 public class ElasticsearchLogRepository extends BatchingLogRepository<XContentBuilder> implements LogRepository, LogSchemaAware {
```
Comment (awislowski) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java]: consider adding Sent to have more descriptive name for this and another ElasticseachLogRepository

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public static final String SENT_TYPE = "sent_messages";
+    public static final String SENT_ALIAS_NAME = "alias_sent_messages";
+    public static final String SENT_TEMPLATE_NAME = "template_sent_messages";
+    public static final String SENT_INDICES_REG_EXP = "sent_messages_*";
+
+    private final Client client;
+    private final FrontendIndexFactory frontendIndexFactory;
+    private final ConsumersIndexFactory consumersIndexFactory;
+
+    public static SchemaManager schemaManagerWithDailIndexes(Client elasticClient) {
```
Comment (awislowski) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java]: typo Daily

Comment (pawelszymczyk) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java]: fixed

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java (hunk 2 above)
Comment (pawelszymczyk) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java]: fixed

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java (hunk 1 above)
Comment (pawelszymczyk) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/consumers/ElasticsearchLogRepository.java]: yeah I know that is little a bit duplication but i think that is better than extending Builders

**Comment by adamdubiel** [comment]
:+1:

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import pl.allegro.tech.hermes.tracker.elasticsearch.consumers.ConsumersIndexFactory;
+import pl.allegro.tech.hermes.tracker.elasticsearch.frontend.FrontendDailyIndexFactory;
+import pl.allegro.tech.hermes.tracker.elasticsearch.frontend.FrontendIndexFactory;
+
+import java.util.concurrent.ExecutionException;
+
+public class SchemaManager {
+
+    public static final String PUBLISHED_INDEX = "published_messages";
+    public static final String PUBLISHED_TYPE = "published_messages";
```
Comment (pjagielski) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java]: it was "published_message" actually ;(

File: hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    public static final String PUBLISHED_ALIAS_NAME = "alias_publishe_messages";
+    public static final String PUBLISHED_TEMPLATE_NAME = "template_published_messages";
+    public static final String PUBLISHED_INDICES_REG_EXP = "published_messages_*";
+
+    public static final String SENT_INDEX = "sent_messages";
+    public static final String SENT_TYPE = "sent_messages";
```
Comment (pjagielski) [inline hermes-tracker-elasticsearch/src/main/java/pl/allegro/tech/hermes/tracker/elasticsearch/SchemaManager.java]: same here

**Comment by pjagielski** [comment]
:+1:

**Pull request merged**